File: C#/OOP/OOP/Car.cs
using System;

namespace OOP {
    abstract class Car {
        public Car() : this("", FuelType.GAS, 0) { }

        public Car(string ownerName, FuelType fuelType, Colour colour)
        {
            _ownerName = ownerName;
            FuelType = fuelType;
            Colour = colour;
            NumberOfCars++;
        }

		private string _ownerName;
		public string OwnerName
        {
            get
            {
                return _ownerName;
            }

			set {
				if (value.Length > 1) {
					_ownerName = value;
				}
			}
		}

		public FuelType FuelType { get; }

		public Colour Colour { get; set; }

		public static int NumberOfCars { get; private set; }

        public virtual void GetSpecialFeature()
        {
            Console.WriteLine("I have no special feature.");
        }

        public abstract void Drive();
	}
}

File: Java/src/Car.java

public abstract class Car {
    private String ownerName;
    private final FuelType fuelType;
    private Colour colour;
    private static int numberOfCars;

    public Car(String ownerName, FuelType fuelType, Colour colour) {
        this.ownerName = ownerName;
        this.fuelType = fuelType;
        this.colour = colour;
        numberOfCars++;
    }

    public Car() {
        this("", FuelType.GAS, Colour.WHITE);
    }

    public String getOwnerName() {
        return ownerName;
    }

    public void setOwnerName(String ownerName) {
        if(ownerName.length() > 1) {
            this.ownerName = ownerName;
        }
    }

    public FuelType getFuelType() {
        return fuelType;
    }

    public Colour getColour() {
        return colour;
    }

    public void setColour(Colour colour) {
        this.colour = colour;
    }

    public static int getNumberOfCars() {
        return numberOfCars;
    }

    private static void setNumberOfCars(int numberOfCars) {
        Car.numberOfCars = numberOfCars;
    }

    public void getSpecialFeature() {
        System.out.println("I have no special feature.");
    }

    public abstract void drive();
}
